### PR TITLE
Low layer

### PR DESCRIPTION
**Pull request by ruengeler**

### CODE REVIEW

File: usrsctplib/user_socket.c [hunk 1]
```diff
@@ -3285,6 +3285,10 @@ usrsctp_conninput(void *addr, const void *buffer, size_t length, uint8_t ecn_bit
 	return;
 }
 
+void usrsctp_fire_timer(int delta)
```
Comment (tuexen) [inline usrsctplib/user_socket.c]: Nit: I would prefer usrsctp_handle_timers() as the function name.
Can we also make the argument an unsigned int?

File: usrsctplib/usrsctp.h [hunk 2]
```diff
@@ -48,6 +48,8 @@ extern "C" {
 #include <netinet/in.h>
 #endif
 
+#include <sys/mbuf.h>
+
```
Comment (tuexen) [inline usrsctplib/usrsctp.h]: I would really like to avoid making the visible to the user...

File: usrsctplib/usrsctp.h [hunk 3]
```diff
@@ -967,6 +969,34 @@ usrsctp_deregister_address(void *);
...
+int
+usrsctp_recv_function_udpsctp4(int, struct mbuf **, struct iovec *);
+
+int
+usrsctp_recv_function_sctp6(int, struct mbuf **, struct iovec *);
+
+int
+usrsctp_recv_function_udpsctp6(int, struct mbuf **, struct iovec *);
+
+
```
Comment (tuexen) [inline usrsctplib/usrsctp.h]: I think all the usrsctp_recv_function() needs to be implemented. Or am I missing something?

Comment (ruengeler) [inline usrsctplib/usrsctp.h]: @tuexen 
The implementation of the usrsctp_recv_functions can be found in the last commit.

**Comment by ruengeler** [comment]
@tuexen 
One more commit to make it compile with THREAD_SUPPORT enabled.

File: usrsctplib/netinet/sctp_pcb.h [hunk 4]
```diff
@@ -322,6 +323,27 @@ struct sctp_base_info {
...
+#if !defined(__Userspace_os_Windows)
+	struct iovec recv_iovec4[MAXLEN_MBUF_CHAIN];
+	struct iovec recv_iovec6[MAXLEN_MBUF_CHAIN];
+	struct iovec udp_recv_iovec4[MAXLEN_MBUF_CHAIN];
+	struct iovec udp_recv_iovec6[MAXLEN_MBUF_CHAIN];
+#else
+    WSABUF *rcv_iovec4;
+    WSABUF *rcv_iovec6;
+    WSABUF *udp_rcv_iovec4;
+    WSABUF *udp_rcv_iovec6;
```
Comment (tuexen) [inline usrsctplib/netinet/sctp_pcb.h]: The indentation seems to be wrong.

**Comment by ruengeler** [comment]
@tuexen 
Indentation fixed

**Pull request closed without merging**